File: go/pkg/models/party_invite.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type PartyInvite struct {
	ID        uuid.UUID `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	InviterID uuid.UUID `json:"inviterId"`
	InviteeID uuid.UUID `json:"inviteeId"`
	Invitee   User      `json:"invitee"`
	Inviter   User      `json:"inviter"`
}

File: dart/unclaimed_streets/lib/models/party_invite.dart
import 'user.dart';

class PartyInvite {
  final String id;
  final String createdAt;
  final String updatedAt;
  final String inviterId;
  final String inviteeId;
  final User invitee;
  final User inviter;

  const PartyInvite({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.inviterId,
    required this.inviteeId,
    required this.invitee,
    required this.inviter,
  });

  factory PartyInvite.fromJson(Map<String, dynamic> json) {
    return PartyInvite(
      id: json['id'] as String,
      createdAt: json['createdAt'] as String? ?? '',
      updatedAt: json['updatedAt'] as String? ?? '',
      inviterId: json['inviterId'] as String,
      inviteeId: json['inviteeId'] as String,
      invitee: User.fromJson(json['invitee'] as Map<String, dynamic>),
      inviter: User.fromJson(json['inviter'] as Map<String, dynamic>),
    );
  }
}
